### PR TITLE
Multiple Enhancements

### PR DESCRIPTION
**Pull request by sbroenne**
- Enable & configure GitGub dependabot
- Add title field
- Add switch to store the contents of the body without any changes
- Updated readme
- Sample index definition json
- Moved web page into its own class

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "" # See documentation for possible values
```
Comment (thomas11) [inline .github/dependabot.yml]: According to the docs, `package-ecosystem` is required. I think `nuget` would be the right value?

File: README.md [hunk 2]
```diff
@@ -1,29 +1,27 @@
 # About
 
-[Azure Search](https://azure.microsoft.com/en-us/services/search/) is a cloud search service for web and mobile app development. This project helps you get content from a website into an Azure Search index. It uses [Abot](https://github.com/sjdirect/abot) to crawl websites. For each page it extracts the content in a customizable way and indexes it into Azure Search.
+[Azure AI Search](https://azure.microsoft.com/en-us/products/ai-services/ai-search/) delivers accurate, hyper-personalized responses in your Gen AI applications. This project helps you get content from a website into an Azure AI Search index. It uses [Abot](https://github.com/sjdirect/abot) to crawl websites. For each page it extracts the content in a customizable way and indexes it into Azure Search.
 
 This project is intended as a demo or a starting point for a real crawler. At a minimum, you'll want to replace the console messages with proper logging, and customize the text extraction to improve results for your use case.
 
-
-# Howto: quick start
+## Howto: quick start
```
Comment (thomas11) [inline README.md]: I don't understand the changed header nesting: the "Howto"'s are not part of "About" IMO.

File: README.md [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
 
 - Create an Azure Search search service. If you're new to Azure Search, follow [this guide](https://docs.microsoft.com/en-us/azure/search/search-create-service-portal).
-- Create an index in your search service with three string fields: "id", "url", and "content". Make them searchable.
-- Run CrawlerMain, either from Visual Studio after opening the .sln file, or from the command line after compiling using msbuild. You will need to pass a few command-line arguments, such as your search service endpoint information (Url) and the root URL of the site you'd like to crawl. Calling the program without arguments or with -h will list the arguments.
-
+- Create an index in your search service with three string fields: "id", "url", "title" and "content". Make them searchable. You can also use the [sample definition](./index.json) as a starter. I have configured the title and content fields to use the "English-analyzer.
```
Comment (thomas11) [inline README.md]:
```suggestion
- Create an index in your search service with three string fields: "id", "url", "title" and "content". Make them searchable. You can also use the [sample definition](./index.json) as a starter. I have configured the title and content fields to use the "English" analyzer.
```

File: .github/dependabot.yml (hunk 1 above)
Comment (sbroenne) [reply .github/dependabot.yml]: Done

File: README.md (hunk 2 above)
Comment (sbroenne) [reply README.md]: Markdownlint complains about multiple H1 in a document but I have reverted the change now.

Comment (thomas11) [reply README.md]: Ah, didn't know that. You could just make them all H2 I guess.

**Review by thomas11 — APPROVED** [review]

**Pull request merged**